Source/JavaScriptCore/jit/JITInlines.h

```diff
@@ -477,6 +477,44 @@ ALWAYS_INLINE ECMAMode JIT::ecmaMode<OpPutPrivateName>(OpPutPrivateName)
     return ECMAMode::strict();
 }
 
+template<size_t minAlign, typename Bytecode>
+ALWAYS_INLINE MacroAssembler::Address JIT::computeGoodBaseAddressForMetadata(const Bytecode& bytecode, GPRReg metadataGPR)
+{
+    // This function attempts to decide what the best base address is when
+    // loading fields from a bytecode instruction's metadata. If offsets are
+    // small enough, we want to emit a single load directly offset from the
+    // metadata table register. But if they're too big, we want to materialize
+    // the metadata offset of the current instruction into a register only
+    // once, so repeated loads don't need to redo that arithmetic.
+
+    // Note: minAlign is very important to get right, but hard to check for
+    // automatically. It should be the minimum alignment or access size across
+    // all the operations we use this address for. This is critical on ARM64
+    // (where this function is most useful) because we have a substantially
+    // larger addressible range when we can assume the access is aligned to
+    // the access size. One important note is that we assume that any accesses
+    // larger than the specified minAlign are also aligned - only then is it
+    // the case that an offset being encodable at a smaller access size also
+    // implies that it is encodable at a larger access size.
+
+    using Metadata = typename Bytecode::Metadata;
+    static_assert(alignof(Metadata) >= minAlign);
```

Comment:
I presume `alignof(Metadata)` have to be aligned on `minAlign` as well?  If so, this would be the better assert: `static_assert(WTF::roundUpToMultipleOf<minAlign>(alignof(Metadata)) == alignof(Metadata));`.